src/handlers/permissions-revoke.ts

```diff
@@ -83,13 +84,12 @@ export class PermissionsRevokeHandler implements MethodHandler {
     }
 
     // Store incoming PermissionsRevoke
-    const indexes: { [key: string]: string } = {
-      interface          : DwnInterfaceName.Permissions,
-      method             : DwnMethodName.Revoke,
-      permissionsGrantId : message.descriptor.permissionsGrantId,
-    };
+    const indexes = PermissionsRevokeHandler.constructMessageStoreIndexes(permissionsRevoke);
     await this.messageStore.put(tenant, message, indexes);
-    await this.eventLog.append(tenant, await Message.getCid(message));
+
+    // get additional indexes for the associated grant message.
+    const additionalIndexes = await PermissionsRevokeHandler.constructAdditionalEventLogIndexes(permissionsGrantMessage);
+    await this.eventLog.append(tenant, await Message.getCid(message), { ...indexes, ...additionalIndexes });
```

Comment:
I think I follow the pattern now, question:

Do you think if there is harm in the message store having the exact same indexes as the Events Store apart from the "additional" storage space used?

You'll probably counter with some questions also, but let's start with this question!